libs/core/calendar/calendar-legend/calendar-legend.component.ts

```diff
@@ -0,0 +1,93 @@
+import {
+    AfterContentInit,
+    Component,
+    ContentChildren,
+    ElementRef,
+    Input,
+    OnInit,
+    QueryList,
+    ViewContainerRef,
+    input
+} from '@angular/core';
+import { SpecialDayRule } from '@fundamental-ngx/core/shared';
+import { CalendarLegendFocusingService } from './calendar-legend-focusing.service';
+import { LegendItemComponent } from './calendar-legend-item.component';
+
+@Component({
+    selector: 'fd-calendar-legend',
+    standalone: true,
+    template: ` <ng-content></ng-content> `,
+    host: {
+        class: 'fd-calendar-legend',
+        '[class.fd-calendar-legend--auto-column]': 'col',
+        '[attr.data-calendar-index]': 'calIndex'
+    }
+})
+export class CalendarLegendComponent<D> implements OnInit, AfterContentInit {
+    /** Get all legend Items */
+    @ContentChildren(LegendItemComponent, { descendants: true })
+    legendItems: QueryList<LegendItemComponent>;
+
+    /** Special
+     * days rules to be displayed in the legend */
+    @Input() specialDaysRules: SpecialDayRule<D>[] = [];
+
+    /**
+     * Make it a column instead
+     */
+    @Input() col = false;
+
+    /** Calendar's index */
+    calIndex: number;
+
+    /** Element getting focused */
+    focusedElement = input<string>('');
+
+    /** @hidden */
+    constructor(
+        private elementRef: ElementRef,
+        private viewContainer: ViewContainerRef,
+        private focusingService: CalendarLegendFocusingService
+    ) {
+        this.calIndex = this.focusingService.getCalIndex() - 1;
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._addCalendarLegend();
```

Comment:
This logic being placed in the init lifecycle method will lead to problems if the special days are updated after the calendar is created. You could watch for changes to the specialDaysRules. But I think a better way of handling this would be to avoid typescript-side logic wherever possible, instead opting to render as much as possible with double curly brace interpolation in `calendar.component.html`